staging/src/k8s.io/apiserver/pkg/server/options/encryption_provider_config_test.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"io/ioutil"
+	"os"
+
+	"k8s.io/apiserver/pkg/storage/value"
+
+	"testing"
+)
+
+var correctConfig string = `
+- kind: AEAD
+  version: v1
+  keys:
+    - name: key1
+      secret: c2VjcmV0IGlzIHNlY3VyZQ==
+    - name: key2
+      secret: dGhpcyBpcyBwYXNzd29yZA==
+  resource: /registry/namespaces
+- kind: AEAD
+  version: v1
+  keys:
+    - name: key2
+      secret: dGhpcyBpcyBwYXNzd29yZA==
+    - name: key3
+      secret: azhzIHNlY3JldCBzdG9yZQ==
+  resource: /registry/
```

Comment:
We shouldn't be exposing etcd keys in this config.  This config should be able kubernetes resource types.  `pkg/storage/storage_factory.go` should accept a group resource override that includes a transformer - that transformer should be global.

I don't see a strong reason to do anything more complicated than identifying specific resources to encrypt (by opaquely passing in an encryption provider).  We deliberately do not expose details of storage location because they are mutable.